src/Microsoft.DotNet.Darc/DarcLib/IGitRepoCloner.cs

```diff
@@ -27,5 +29,12 @@ public interface IGitRepoCloner
     /// <param name="repoUri">Repository uri to clone</param>
     /// <param name="targetDirectory">Target directory to clone to</param>
     /// <param name="gitDirectory">Location for the .git directory, or null for default</param>
-    public void Clone(string repoUri, string targetDirectory, string? gitDirectory);
+    public Task CloneAsync(string repoUri, string targetDirectory, string? gitDirectory);
```

Comment:
The summary in the documentation in both overrides seems different (clone without checking out the working tree vs just clone). This creates expectation that the behavior of both overrides might differ. Is that correct? If so, it should be reflected in the name.